service/controller/resource/namespace/current.go

```diff
@@ -72,18 +72,29 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	// we get an empty list here after the delete event got replayed. Then we just
 	// remove the namespace as usual.
 	if key.IsDeleted(&customObject) {
-		n := key.ClusterNamespace(customObject)
+		namespace := key.ClusterNamespace(customObject)
 
-		lo := metav1.ListOptions{
+		deployments, err := r.k8sClient.AppsV1().Deployments(namespace).List(ctx, metav1.ListOptions{
 			LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, key.OperatorName),
+		})
+		if err != nil {
+			return nil, microerror.Mask(err)
 		}
 
-		list, err := r.k8sClient.AppsV1().Deployments(n).List(ctx, lo)
+		// Ensure PVCs have been deleted so that bound PVs are properly cleaned up.
```

Comment:
When a cluster is being deleted, we generally rely on namespace deletion to clean up everything. This was preventing our PVC deletion logic from running.